lib/cache/resource.js

```diff
@@ -0,0 +1,122 @@
+'use strict';
+
+var clone = require('clone');
+var MemoryCache = require('./memory');
+
+function ResourceCache(cacheId, options) {
+    this._cache = new MemoryCache(cacheId, options);
+}
+
+ResourceCache.prototype.get = function(cacheKey) {
+    return getFromCache.call(this, cacheKey);
+};
+
+function getFromCache(cacheKey) {
+    var entry = clone(this._cache.get(cacheKey));
```

Comment:
When we get the resource and swap out the cached nested ones for the actual values stored in them it'd keep the actual values instead of the cache references. The downside of this is mainly in handling expiration. Let's say there's a nested resource and it hasn't expired yet and i ask to get it. If I don't clone it'll update the nested resource cache reference with the actual value and the next time a call a get on the same resource and the nested resource has expired, it'll give me the resource, when the proper behavior would be to give me undefined since the nested resource has expired.